app/src/main/java/org/mozilla/fenix/tabstray/TrayViewHolders.kt

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray
+
+import android.view.View
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.extensions.LayoutContainer
+import org.mozilla.fenix.R
+import org.mozilla.fenix.tabstray.browser.BaseBrowserTrayList
+
+sealed class TrayViewHolder constructor(
+    override val containerView: View
+) : RecyclerView.ViewHolder(containerView), LayoutContainer {
+
+    abstract fun bind(adapter: RecyclerView.Adapter<out RecyclerView.ViewHolder>)
+}
+
+class BrowserTabViewHolder(
+    containerView: View,
+    interactor: TabsTrayInteractor
+) : TrayViewHolder(containerView) {
+
+    private val trayList: BaseBrowserTrayList = itemView.findViewById(R.id.tray_list_item)
+
+    init {
+        trayList.interactor = interactor
```

Comment:
@Amejia481 this is how we're passing the interactor to the viewholders. Keen if you have other suggestions how to do this. 🙂

Reply:
I think it's OK!